### PR TITLE
Updating gbmv, gemv, ger, hbmv

### PR DESCRIPTION
**Pull request by daineAMD**
This PR adds gbmv, gemv, ger, and hbmv to hipblas bench. It also adds device_mode tests, updates to use the new handle, and updates to use more recent batched arrays, along with any other cleanup I noticed.

Working my way through all the functions, so expect more of these PRs. Have a couple L1s left, and a majority of L2s, and all L3s.

### CODE REVIEW

File: clients/gtest/gemv_strided_batched_gtest.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ TEST_P(gemv_gtest_strided_batched, gemv_gtest_float)
 {
     Arguments arg = setup_gemv_arguments(GetParam());
 
-    hipblasStatus_t status = testing_gemvStridedBatched<float>(arg);
+    hipblasStatus_t status = testing_gemv_strided_batched<float>(arg);
```
Comment (TorreZuk) [inline clients/gtest/gemv_strided_batched_gtest.cpp:161]: Sorry if I followed that old convention too.

File: clients/include/testing_gbmv_batched.hpp [hunk 2]
```diff
@@ -66,143 +62,154 @@ hipblasStatus_t testing_gbmvBatched(const Arguments& argus)
...
+    hipblasLocalHandle handle(argus);
 
-    double gpu_time_used, cpu_time_used;
-    double hipblasGflops, cblas_gflops, hipblasBandwidth;
-    double rocblas_error;
+    double gpu_time_used, hipblas_error_host, hipblas_error_device;
 
-    T alpha = (T)argus.alpha;
-    T beta  = (T)argus.beta;
+    T h_alpha = (T)argus.alpha;
```
Comment (TorreZuk) [inline clients/include/testing_gbmv_batched.hpp:69]: Nan testing something for the future, but could use accessor even before we try it.  A different PR.

File: clients/include/testing_gbmv_strided_batched.hpp [hunk 3]
```diff
@@ -97,47 +98,62 @@ hipblasStatus_t testing_gbmvStridedBatched(const Arguments& argus)
...
 
     // copy vector is easy in STL; hz = hy: save a copy in hz which will be output of CPU BLAS
-    hz = hy;
+    hy_cpu = hy;
 
     // copy data from CPU to device
-    hipMemcpy(dA, hA.data(), sizeof(T) * A_size, hipMemcpyHostToDevice);
-    hipMemcpy(dx, hx.data(), sizeof(T) * X_size, hipMemcpyHostToDevice);
-    hipMemcpy(dy, hy.data(), sizeof(T) * Y_size, hipMemcpyHostToDevice);
+    CHECK_HIP_ERROR(hipMemcpy(dA, hA.data(), sizeof(T) * A_size, hipMemcpyHostToDevice));
```
Comment (TorreZuk) [inline clients/include/testing_gbmv_strided_batched.hpp:104]: Another comment not for this PR, but we should have added hA.dataSize() for all these types of transfers.

File: clients/include/testing_gemv_batched.hpp [hunk 4]
```diff
@@ -65,121 +61,111 @@ hipblasStatus_t testing_gemvBatched(const Arguments& argus)
...
+                                                 dx.ptr_on_device(),
                                                  incx,
-                                                 (T*)&beta,
-                                                 dy_array,
+                                                 d_beta,
+                                                 dy.ptr_on_device(),
                                                  incy,
                                                  batch_count));
 
+        CHECK_HIP_ERROR(hy_device.transfer_from(dy));
```
Comment (TorreZuk) [inline clients/include/testing_gemv_batched.hpp:138]: Could be pushed after the cpu reference to hide some of the cpu ref. calculation time.

**Review by TorreZuk — APPROVED** [review]
Very good, comments all for future reference.  Now it really needs a bench test suite added much like rocblas also does.

**Comment by daineAMD** [comment]
Thanks for the comments Torre, I'll make a ticket for generic hipblas test cleanup and add some of your suggestions.

**Pull request merged**